temporal-sdk/src/main/java/io/temporal/internal/sync/DeterministicRunnerImpl.java

```diff
@@ -360,10 +313,10 @@ public void close() {
       return;
     }
     try {
-      for (WorkflowThread c : threadsToAdd) {
+      for (WorkflowThread c : workflowThreadsToAdd) {
         threads.add(c);
       }
-      threadsToAdd.clear();
+      workflowThreadsToAdd.clear();
```

Comment:
Should we do the same for the callback threads to add?

Reply:
It's actually what I did in this PR, I made a collection `callbackThreadsToAdd` that works almost the same way (just traversed in a reversed order when we add)